TMDBApp/TMDBApp/Requests.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  Requests.swift
+//  TMDBApp
+//
+//  Created by Richard Leh on 19/12/2016.
+//  Copyright © 2016 Richard Leh. All rights reserved.
+//
+
+import Foundation
+import Alamofire
+
+// MARK: API TMDB MOVIE
+private struct TMDBMovieAPI {
+    
+    static let key = "28190e24b6d3d0b73a9b5842bee9d8bf" // 1f54bd990f1cdfb230adb312546d765d​
+    
+    static let upcomingPath = "https://api.themoviedb.org/3/movie/upcoming"
+    static let genrerPath   = "https://api.themoviedb.org/3/genre/movie/list"
+    static let searchPath   = "http://api.themoviedb.org/3/search/movie"
+    static let imagePath    = "https://image.tmdb.org/t/p/w780"
+    
+}
+
+enum RequestError: Error {
+    case InvalidRequest(String)
+}
+
+class Requests: NSObject{
+    
+    class func sharedInstance() -> Requests {
+        struct Singleton {
+            static var sharedInstance = Requests()
+        }
+        return Singleton.sharedInstance
+    }
+    
+    fileprivate func request(fromString requestString: String, withCompletion completion: @escaping CompletionResultError) {
+        
+        Alamofire.request(requestString).validate().responseJSON { response in
+            //debugPrint(response)
+            
+            switch response.result {
+            case .success:
+                if let json = response.result.value {
+                    completion(json, nil)
```

Comment:
How could we use the Swift 3 API guidelines to improve the readability of this call? For instance, what is "json" and what is "nil"?